src/backup.py

```diff
@@ -161,6 +176,46 @@ def can_use_bucket(self) -> bool:
             return False
         return True
 
+    def list_backups(self) -> List[S3Backup]:
+        """List the backups stored in S3 in the current s3 configuration.
+
+        Returns:
+            list of backups.
+        """
+        backups = []
+        for item in self._iterate_objects():
+            s3_object_key = pathlib.Path(item["Key"])
+            backup_id = (s3_object_key).relative_to(self._prefix)
+            backup = S3Backup(
+                backup_id=str(backup_id),
+                last_modified=item["LastModified"],
+                size=item["Size"],
+            )
+            backups.append(backup)
+        return backups
+
+    def _iterate_objects(self) -> Generator[dict, None, None]:
```

Comment:
I find the method name a big generic

Reply:
Renamed to `list_s3_objects` that I believe gives it a hint that we are querying S3

Reply:
Thanks